govcd/catalog.go

```diff
@@ -501,8 +514,10 @@ func getExistingCatalogItems(catalog *Catalog) (catalogItemNames []string) {
 // uploadedBytes - how much of file already uploaded
 // fileSizeToUpload - how much bytes will be uploaded
 // uploadPieceSize - size of chunks in which the file will be uploaded to the catalog.
+// uploadedBytesForCallback all uploaded bytes if multi disk in ova
+// allFilesSize overall sum of size if multi disk in ova
 // callBack a function with signature //function(bytesUpload, totalSize) to let the caller monitor progress of the upload operation.
-func uploadFile(client *Client, uploadLink, filePath string, uploadedBytes, fileSizeToUpload int64, uploadPieceSize int64, callBack func(bytesUpload, totalSize int64)) (int64, error) {
+func uploadFile(client *Client, uploadLink, filePath string, uploadedBytes, fileSizeToUpload int64, uploadPieceSize int64, uploadedBytesForCallback, allFilesSize int64, callBack func(bytesUpload, totalSize int64)) (int64, error) {
```

Comment:
`uploadFile` and `uploadPartFile` have definitely too many arguments to pass by position. This makes reading the code unnecessarily difficult.
It would be better to create a structure to pass, using explicit field names. It would improve readability and facilitate debugging.

Reply:
Changed.